### PR TITLE
Updates binary.js to support Node Buffer objects

### PR DESCRIPTION
**Pull request by nzack**
To test:
`$ npm test`

### CODE REVIEW

File: lib/binary.js [hunk 1]
```diff
@@ -164,20 +216,58 @@ var arrayBufferConversion = {
...
+  },
+  buffer: function(arrayBuffer) {
+    // NOTE: Won't work in 0.12.x versions of Node
+    return new Buffer(arrayBuffer);
+  }
+};
+
+var bufferConversion = {
+  arrayBuffer: function(buffer) {
+    debugger;
```
Comment (payaaam) [inline lib/binary.js]: .

File: lib/binary.js [hunk 2]
```diff
@@ -164,20 +216,58 @@ var arrayBufferConversion = {
...
+    if (buffer.buffer) {
+      return buffer.buffer.slice(buffer.byteOffset, buffer.byteOffset + buffer.byteLength);
+    }
+
+    var arrayBuffer = new ArrayBuffer(buffer.length);
+    var bufferView = new Uint8Array(arrayBuffer);
+    for (var i = 0; i < buffer.length; i++) {
+      bufferView[i] = buffer[i];
+    }
+    debugger;
```
Comment (payaaam) [inline lib/binary.js]: .

File: tests/slice-tests.js [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    it(test2.message, function(){ test = test2.test(bin) });
+    it(test3.message, function(){ test = test3.test(bin) });
+    it(test4.message, function(){ test = test4.test(bin) });
+    it(test5.message, function(){ test = test5.test(bin) });
+    it(test6.message, function(){ test = test6.test(bin) });
+    it(test7.message, function(){ test = test7.test(bin) });
+    it(test8.message, function(){ test = test8.test(bin) });
+    it(test9.message, function(){ test = test9.test(bin) });
+    //    it(test10.message, function(){ test = test10.test(bin) });
+    //    it(test11.message, function(){ test = test11.test(bin) });
```
Comment (payaaam) [inline tests/slice-tests.js]: remove if unneeded

**Comment by payaaam** [comment]
can we create a change-log on here?

File: tests/slice-tests.js (hunk 3 above)
Comment (nzack) [inline tests/slice-tests.js]: gone

**Comment by nzack** [comment]
changelog added

**Pull request merged**